src/electionguard/ballot_compact.py

```diff
@@ -0,0 +1,138 @@
+)
+from .ballot_box import BallotBoxState
+from .election import (
+    CiphertextElectionContext,
+    ContestDescriptionWithPlaceholders,
+    InternalElectionDescription,
+)
+from .encrypt import encrypt_ballot_contests
+from .utils import get_optional
+
+
+@dataclass
+class CompactPlaintextBallot:
+    """A compact plaintext representation of ballot minimized for data size"""
+
+    object_id: str
+    style_id: str
+    selections: List[bool]
+    extra_data: Dict[int, ExtendedData]
+
+
+@dataclass
+class CompactSubmittedBallot:
+    """A compact submitted ballot minimized for data size"""
+
+    compact_plaintext_ballot: CompactPlaintextBallot
+    timestamp: int
+    ballot_nonce: int
+    previous_tracking_hash: int
+    tracking_hash: int
+    ballot_box_state: BallotBoxState
+
+
+def expand_compact_submitted_ballot(
+    compact_ballot: CompactSubmittedBallot,
+    description: InternalElectionDescription,
+    context: CiphertextElectionContext,
+) -> SubmittedBallot:
+    """
+    Expand a compact submitted ballot using context and
+    the election description into a submitted ballot
+    """
+    # Convert to Q from int
+    tracking_hash = int_to_q_unchecked(compact_ballot.tracking_hash)
+    previous_tracking_hash = int_to_q_unchecked(compact_ballot.previous_tracking_hash)
+    ballot_nonce = int_to_q_unchecked(compact_ballot.ballot_nonce)
+
+    # Expand ballot and encrypt & hash contests
+    plaintext_ballot = expand_compact_plaintext_ballot(
+        compact_ballot.compact_plaintext_ballot, description
+    )
+    contests = get_optional(
+        encrypt_ballot_contests(plaintext_ballot, description, context, ballot_nonce)
+    )
+    crypto_hash = create_ballot_hash(
+        plaintext_ballot.object_id, description.description_hash, contests
+    )
+
+    return SubmittedBallot(
+        plaintext_ballot.object_id,
+        plaintext_ballot.ballot_style,
+        description.description_hash,
+        previous_tracking_hash,
+        contests,
+        tracking_hash,
+        compact_ballot.timestamp,
+        crypto_hash,
+        ballot_nonce,
+        compact_ballot.ballot_box_state,
+    )
+
+
+def expand_compact_plaintext_ballot(
+    compact_ballot: CompactPlaintextBallot, description: InternalElectionDescription
+) -> PlaintextBallot:
+    """Expand a compact plaintext ballot into the original plaintext ballot"""
+    return PlaintextBallot(
+        compact_ballot.object_id,
+        compact_ballot.style_id,
+        _get_plaintext_contests(compact_ballot, description),
+    )
+
+
+def _get_plaintext_contests(
+    compact_ballot: CompactPlaintextBallot, description: InternalElectionDescription
+) -> List[PlaintextBallotContest]:
+    """Get ballot contests from compact plaintext ballot"""
+    index = 0
+    ballot_style_contests = _get_ballot_style_contests(
+        compact_ballot.style_id, description
+    )
+
+    contests: List[PlaintextBallotContest] = []
+    for description_contest in sorted(
```

Comment:
This sort I feel like we don't often do, but I'm ensuring that it is used. Perhaps this should be a utility function.